blog/content/post/2020-06-30-trenchboot-grub-cbfs.md

```diff
@@ -0,0 +1,575 @@
+---
+title: "TrenchBoot: Open Source DRTM. GRUB's new features and TPM event log."
+abstract: This blog post will show you what features we have added to GRUB
+          and why they are useful from user's point of view. Also, there
+          will be shown how to utilize TPM event logs and hence debug DRTM.
+cover: /covers/trenchboot-logo.png
+author: piotr.kleinschmidt
+layout: post
+published: true
+date: 2020-07-03
+archives: "2020"
+
+tags:
+  - security
+  - open-source
+  - grub
+  - trenchboot
+  - nixos
+categories:
+  - Firmware
+  - Security
+
+---
+
+If you haven't read previous blog posts from *TrenchBoot* series, we strongly
+encourage to catch it up. Best way, is to search under
+[TrenchBoot](https://blog.3mdeb.com/tags/trenchboot/) tag. In this article, we
+want to show what changes have been made in GRUB and what was our motivation to
+introduce them. Also, we will introduce extension of **TPM event log** which
+lets users  debug DRTM and verify some operations. Besides theoretical
+considerations, there is the **verification part** which introduce particular
+examples and verification procedures.
+
+## GRUB additional feature - CBFS and LZMA support
+
+Before moving to essential part of GRUB development, first you need to
+understand what stands behind **CBFS** and **LZMA**. *CBFS* stands for *coreboot
+filesystem* and it is 'kind of layout' in which every coreboot's binary file is
+organized. As name might be misleading, it is **not actual filesystem**. It only
+describes what components in what order should be presented in SPI Flash (if it
+is flashed with coreboot of course). With such unification, it is much easier
+for developers and users to manipulate coreboot's image, debug it or check its
+integrity. Also, it allows to makes changes on final binary file (e.g. adding or
+removing some part of coreboot) by utilizng special tool - `cbfstool`. More
+about it will be presented later.
+
+Second abbreviation - **LZMA** stands for *Lempel-Ziv-Markov chain-Algorithm*
+and it is **compression algorithm** strongly used in coreboot (and not only
+there). As SPI flash capacity is not great, but hardware and BIOS is getting
```

Comment:
I would also mention relatively low read speeds of flash, so less data to read = faster boot, even with decompression time, which for LZMA is one of the worst when compared to other algorithms.

Reply:
Added information about boot time speed up.